examples/skim.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+import json, re, os
+import ROOT
+
+class JSONFileFilter:
+    def __init__(self,fname):
+        self.keep = {}
+        for _run,lumis in json.load(open(fname, 'r')).iteritems():
+            run = long(_run)
+            if run not in self.keep: self.keep[run] = []
+            self.keep[run] += lumis
+    def filterRunLumi(self,run,lumi):
+        try:
+            for (l1,l2) in self.keep[run]:
+                if l1 <= lumi and lumi <= l2: return True
+            return False
+        except KeyError:
+            return False
+    def filterRunOnly(self,run):
+        return (run in self.keep)
+
+def _runIt(args):
+    
+    fname,outdir,suffix,drops,keeps,cutstring,selector,justcount = args
+
+    infile = ROOT.TFile(fname)
+    intree = infile.Get('Events')
+    lumitree = infile.Get('LuminosityBlocks')
+    runtree = infile.Get('Runs')
+
+    for drop in drops: intree.SetBranchStatus(drop,0)
+    for keep in keeps: intree.SetBranchStatus(keep,1)
+
+    if fname[-5:]!='.root': raise RuntimeError
+    if not justcount:
+        outfile = ROOT.TFile('%s/%s%s.root'%(outdir,fname.split('/')[-1][:-5],suffix),'recreate','',9)
+        outfile.cd()
+
+    jsonelist = None
+    if selector:
+        intree.SetBranchStatus('run',1)
+        intree.SetBranchStatus('luminosityBlock',1)
+        jsonelist = ROOT.TEntryList('jsonelist','jsonelist')
+        n = long(0)
+        for ev in intree:
+            if selector.filterRunLumi(ev.run,ev.luminosityBlock): jsonelist.Enter(n)
```

Comment:
actually the question is more relevant here: how fast is it here?
would it make sense to apply first the cut string and then the json? (the cutstring could be much more selective than the json)